src/nvim/version_defs.h

```diff
@@ -41,16 +43,5 @@
 // for the ":version" command and "nvim -h"
 #define NVIM_VERSION_LONG "NVIM " NVIM_VERSION_MEDIUM "." STR(NVIM_VERSION_PATCH) NVIM_VERSION_PRERELEASE NVIM_VERSION_BUILD
 
-//
-// Vim version number, name, etc. Patchlevel is defined in version.c.
-//
-#define VIM_VERSION_MAJOR                7
-#define VIM_VERSION_MINOR                4
-#define VIM_VERSION_100     (VIM_VERSION_MAJOR * 100 + VIM_VERSION_MINOR)
-
-// used for the runtime directory name
-#define VIM_VERSION_NODOT       "vim74"
-// swap file compatibility (max. length is 6 chars)
-#define VIM_VERSION_SHORT       "7.4"
 
 #endif  // NVIM_VERSION_DEFS_H
```

Comment:
You're right, the naming is a bit ad-hoc right now. Perhaps `version_gen.h` or something. The emphasis should be that it is re-generated while `version.h` is not.